test/sde/sde_dynamical.jl

```diff
@@ -0,0 +1,31 @@
+using StochasticDiffEq, DiffEqNoiseProcess, Test, DiffEqDevTools, Random
+Random.seed!(1)
+
+u0 = zeros(2)
+v0 = ones(2)
+γ = 50
+
+f1_harmonic(v,u,p,t) = -u
+f2_harmonic(v,u,p,t) = v
+g(u,p,t) = sqrt(2γ)
+
+f1_harmonic_iip(dv,v,u,p,t) = dv .= f1_harmonic(v,u,p,t)
+f2_harmonic_iip(du,v,u,p,t) = du .= f2_harmonic(v,u,p,t)
+g_iip(du,u,p,t) = du .= g(u,p,t)
+
+ff_harmonic = DynamicalSDEFunction(f1_harmonic,f2_harmonic,g)
+prob1 = DynamicalSDEProblem(ff_harmonic,g,v0,u0,(0.0,5.0))
+sol1 = solve(prob1,BAOAB(gamma=γ);dt=1/10,save_noise=true)
+
+prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
+sol2 = solve(prob2,BAOAB(gamma=γ);dt=1/10)
+
+@test sol1[:] ≈ sol2[:]
+
+dts = (1/2) .^ (9:-1:3)
+# Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
+# I don't think there are any analytic solutions for problems of this type
+sim1  = analyticless_test_convergence(dts,prob1,BAOAB(gamma=γ),(1/2)^10;trajectories=Int(1e2),use_noise_grid=false)
```

Comment:
That's definitely not enough trajectories to produce a weak convergence estimate. Do a local plot with like 5e5 trajectories and show me the `plot(sim1)`.

Reply:
![errors](https://user-images.githubusercontent.com/38594562/107847895-393b6980-6de7-11eb-94d3-88418128295a.png)

Here's the plot with 5e4 trajectories with tspan=(0.0,0.5). I seemed to be running into memory issues if I try more trajectories or a longer tspan. However, it does seem to be converged now. I also tried with 1e4 trajectories and the plot is only marginally different.

Here's the estimate I obtain:
```julia
julia> sim1.𝒪est
Dict{Any,Any} with 2 entries:
  :weak_final => 1.45594
  :final      => 0.802602
```

Reply:
What if you make the dts a bit smaller? the strong convergence estimate seems to not entire its asymtopic regime until half way through.

Reply:
![errors](https://user-images.githubusercontent.com/38594562/107854675-6f8fdd80-6e15-11eb-8808-2802a718f1cd.png)

This is with only 1e4 trajectories as it takes a very long time with smaller timesteps. The test dt here is 0.5^12.
```julia
julia> sim1.𝒪est
Dict{Any,Any} with 2 entries:
  :weak_final => 1.19034
  :final      => 0.777283
```

This doesn't quite match up with the previous plot however so I guess it's not properly converged. I'm not sure I can run it for many more trajectories though.

Reply:
More trajectories won't help. You'd see that because it would flatline in a noisy fashion on the left end due to not having enough values for the sampling error. This doesn't show sampling eror. It shows 1st order convergence.

Reply:
Lower `dt`s would be the check to get rid of that wobble though.

Reply:
Oh ok, does this mean the algorithm is 1st order then? I don't know much about SDEs, I was just going off the plots in the paper, but that's converging observables calculated from the distributions obtained from the integration. Is that different?

Reply:
A lot of the convergence diagrams aren't well-behaved in their paper either. Can you try and recreate one and see if it's around the same?

Reply:
It turns out I had made a mistake in the algorithm which I've now fixed. The convergence is a lot more well behaved now and I was able to roughly reproduce one of their plots.
![image](https://user-images.githubusercontent.com/38594562/108986015-99ed6080-7689-11eb-8b7c-697c9ef54119.png)
![image](https://user-images.githubusercontent.com/38594562/108986043-a07bd800-7689-11eb-8ca5-9bdd77890964.png)

I did 2000 trajectories with 1e6 steps and they did 2000 with 1e8 so I think it's safe to assume I'd converge to the same result given enough timesteps.